src/tools/pkcs11-tool.c

```diff
@@ -1163,6 +1163,15 @@ static int login(CK_SESSION_HANDLE session, int login_type)
 
 	if (!pin && !(info.flags & CKF_PROTECTED_AUTHENTICATION_PATH)) {
 			printf("Logging in to \"%s\".\n", p11_utf8_to_local(info.label, sizeof(info.label)));
+		if ((login_type == CKU_SO && info.flags & CKF_SO_PIN_COUNT_LOW) ||
+				(login_type == CKU_USER && info.flags & CKF_USER_PIN_COUNT_LOW))
```

Comment:
That looks fine to me. I also added the CKF_SO_PIN* flags user readable strings in case that was the way to go. Since there is no other comments at the moment, I am updating the commits with your suggestion.